x-pack/plugins/ml/public/application/components/job_spaces_list/job_spaces_list.tsx

```diff
@@ -66,7 +66,9 @@ export const JobSpacesList: FC<Props> = ({ spacesApi, spaceIds, jobId, jobType,
     });
   }
 
-  const { SpaceList, ShareToSpaceFlyout } = spacesApi.ui.components;
+  const LazySpaceList = React.lazy(spacesApi.ui.components.getSpaceList);
+  const LazyShareToSpaceFlyout = React.lazy(spacesApi.ui.components.getShareToSpaceFlyout);
```

Comment:
There isn't really anything 'outside' of `render` in a FC 😅 . But `useMemo` should be used around the `React.lazy` declaration, yea (as it was properly done in `saved_objects_table_page.tsx`)

Reply:
Wherever I could do so (inside the Spaces plugin) I pulled out the lazy loading to the top level.
Otherwise, I memoized these inside of function components.